doc/hardware/emulator/adc_emul.rst

```diff
@@ -0,0 +1,24 @@
+.. _adc_emul_api:
+
+
+ADC Emulator
+###################################
+
+Overview
+********
+The ADC emulator provides the ability to be perform tests without access to
+the real hardware.
+
+Configuration Options
+*********************
+
+Related configuration options:
+
+* :kconfig:option:`CONFIG_ADC_EMUL`
```

Comment:
You might want to mention the configs for the corresponding drivers, e.g. `CONFIG_ADC`.

Reply:
Force-pushed the changes to add these extra options.